nodejs/week2/app.js

```diff
@@ -0,0 +1,77 @@
+/** @format */
+
+const { query} = require('express');
+const express = require('express');
+const { string } = require('yargs');
+const app = express();
+const port = process.env.PORT || 3000;
+const  documents= require('./documents.json');
+
+app.use(express.json());
+
+app.get('/', (req, res) => {
+  res.send('This is a search engine');
+});
+
+app.get('/search', async(req, res) => {
+  try{
+  const q = res.query.q;
+  if (q) {
+    const fitterdq = documents.filter((document) => document.includes(q));
+   
+    res.json(fitterdq);
+  } else {
+    res.json(documents);
+  }
+}catch (error) {
+  res.status(500).json({error:"internal server error!"});
+}
+});
```

Comment:
The functions used in these endpoints do not return a promise. We therefore do not need to await anything.
You can leave out the async/await and try/catch :ok_hand: 


In the filter function `document` is an object. The `includes` function works for arrays. The filter function will look at the objects in the array one by one.

I would suggest looking at [Object.entries()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) or [Object.values()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values)

Let me know if anything needs more explanation!